### PR TITLE
Enables Waypoint Integrations

### PR DESCRIPTION
**Pull request by BrandonRomano**
[🔍 Preview](https://dev-portal-git-waypoint-integrations-hashicorp.vercel.app/waypoint/integrations)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203910916598953
  - https://app.asana.com/0/0/1203697855994216

### CODE REVIEW

File: src/lib/content-adjustments/rewrite-waypoint-plugins-to-integrations.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Rewrite Waypoint `plugin` links to `integrations`.
+ *
+ * Waypoint /plugins have been moved to /integrations. We have redirects
+ * set up to handle this, and we hope to update these links in content,
+ * but as a temporary solution we can rewrite these links at compile time.
+ *
+ * Note: we intentionally run this _after_ URLs have been normalized to
+ * include their product prefix. This way, regardless of whether Waypoint
+ * content authors are writing `/plugins` or `/waypoint/plugins`, we should
+ * apply the `plugins` → `integrations` rewrites correctly.
```
Comment (BRKalow) [inline src/lib/content-adjustments/rewrite-waypoint-plugins-to-integrations.ts:11]: This is a helpful note!

File: build-libs/redirects.js [hunk 2]
```diff
@@ -281,6 +281,23 @@ async function buildDevPortalRedirects() {
...
+		{
+			source: '/waypoint/plugins',
+			destination: '/waypoint/integrations',
+			permanent: true,
+		},
+		{
+			source: '/waypoint/plugins/:slug',
+			destination: '/waypoint/integrations/hashicorp/:slug',
+			permanent: true,
+		},
```
Comment (BRKalow) [inline build-libs/redirects.js:300]: 👍

**Review by BRKalow — APPROVED** [review]
LGTM! :shipit:

**Review by kylemac — APPROVED** [review]

**Review by thiskevinwang — APPROVED** [review]
✅ 

Tested one redirect — appears to be working 👍 
- https://dev-portal-git-waypoint-integrations-hashicorp.vercel.app/waypoint/plugins/aws-ssm
- https://dev-portal-git-waypoint-integrations-hashicorp.vercel.app/waypoint/integrations/hashicorp/aws-ssm

**Pull request merged**